### ISSUE TITLE
Background document GC does not update bucket DB atomically, causing false positives for merge pending metrics

### ISSUE THREAD

**Issue by vekterli** [vespa-engine/vespa#11373]
Background document GC is implemented in terms of `RemoveSelection` ops sent in parallel to bucket replicas. These are low priority and may be stuck in persistence queues for a while, depending on the load on the replica nodes.

Today, when a reply is received on the distributor, the GC maintenance operation will immediately update the bucket DB with the information returned as part of the reply. If a periodic DB metric updating scan happens to see this bucket in a state where not all replies have been received, it will likely be counted as a bucket requiring a merge. This causes confusing merge pending metric noise in otherwise quiescent systems.

GC ops (and other ops, for that matter) should preferably update the bucket DB as one atomic operation when all replies have been received. This needs some extra care to make sure doing so does not overwrite any more recent bucket DB updates received as part of concurrent feed load.
